ygot/diff.go

```diff
@@ -242,6 +245,19 @@ func findSetLeaves(s GoStruct, opts ...DiffOpt) (map[*pathSpec]interface{}, erro
 		if err != nil {
 			return util.NewErrs(err)
 		}
+		//prevent duplicate key
+		keys := make([]string, len(vp.gNMIPaths))
+		for i, paths := range vp.gNMIPaths {
+			s, _ := PathToString(paths)
```

Comment:
Can we catch the error cleanly here? I think that if we find an error here we can just return that this wasn't possible -- rather than crashing or potentially silently failing.